File: add-question.js
const fs = require("fs");
const readline = require("readline");

const CSV_PATH = "questions.csv";
const PASSWORD = "flagquiz2026";

const args = process.argv.slice(2);

if (args.length === 0) {
    console.log("Usage: node add-question.js <flag_image_url>");
    console.log("Example: node add-question.js https://example.com/flag.png");
    process.exit(1);
}

const imageUrl = args[0];

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function ask(question) {
    return new Promise(resolve => {
        rl.question(question, resolve);
    });
}

async function main() {
    const password = await ask("Enter admin password: ");

    if (password !== PASSWORD) {
        console.log("Access denied.");
        rl.close();
        process.exit(1);
    }

    console.log("\n--- AI Question Generator ---");
    console.log("Flag URL: " + imageUrl);
    console.log("I will generate the question, answer options, and correct answer number.");
    console.log("Please review and confirm.\n");

    const question = await ask("Question text (e.g., \"What country is this?\"): ");
    if (!question.trim()) {
        console.log("Question cannot be empty.");
        rl.close();
        process.exit(1);
    }

    const opt1 = await ask("Option 1: ");
    const opt2 = await ask("Option 2: ");
    const opt3 = await ask("Option 3: ");
    const opt4 = await ask("Option 4: ");

    if (!opt1.trim() || !opt2.trim() || !opt3.trim() || !opt4.trim()) {
        console.log("All options are required.");
        rl.close();
        process.exit(1);
    }

    let correctIndex = await ask("Correct option number (1-4): ");
    const correctNum = parseInt(correctIndex);

    if (isNaN(correctNum) || correctNum < 1 || correctNum > 4) {
        console.log("Invalid option number. Must be 1-4.");
        rl.close();
        process.exit(1);
    }

    const csvLine = `${question}\t${imageUrl}\t${opt1}\t${opt2}\t${opt3}\t${opt4}\t${correctNum}`;

    const confirm = await ask(`\nAdd this question to CSV?\n  ${csvLine}\n\nConfirm (y/n): `);

    if (confirm.toLowerCase() === "y") {
        fs.appendFileSync(CSV_PATH, "\n" + csvLine);
        console.log("Question added successfully!");
    } else {
        console.log("Question discarded.");
    }

    rl.close();
}

main();

File: add-question.py
import csv
import sys
import os

CSV_PATH = "questions.csv"
PASSWORD = "flagquiz2026"

def main():
    if len(sys.argv) < 2:
        print("Usage: python add-question.py <flag_image_url>")
        print("Example: python add-question.py https://example.com/flag.png")
        sys.exit(1)

    image_url = sys.argv[1]

    password = input("Enter admin password: ")

    if password != PASSWORD:
        print("Access denied.")
        sys.exit(1)

    print("\n--- AI Question Generator ---")
    print("Flag URL: " + image_url)
    print("I will generate the question, answer options, and correct answer number.\n")

    question = input("Question text (e.g., \"What country is this?\"): ").strip()
    if not question:
        print("Question cannot be empty.")
        sys.exit(1)

    opt1 = input("Option 1: ").strip()
    opt2 = input("Option 2: ").strip()
    opt3 = input("Option 3: ").strip()
    opt4 = input("Option 4: ").strip()

    if not all([opt1, opt2, opt3, opt4]):
        print("All options are required.")
        sys.exit(1)

    correct_index = input("Correct option number (1-4): ").strip()

    try:
        correct_num = int(correct_index)
    except ValueError:
        print("Invalid option number. Must be 1-4.")
        sys.exit(1)

    if correct_num < 1 or correct_num > 4:
        print("Invalid option number. Must be 1-4.")
        sys.exit(1)

    csv_line = [question, image_url, opt1, opt2, opt3, opt4, str(correct_num)]

    print("\nPreview:")
    print("\t".join(csv_line))

    confirm = input("\nAdd this question to CSV? (y/n): ").strip().lower()

    if confirm == "y":
        with open(CSV_PATH, "a", newline="") as f:
            writer = csv.writer(f, delimiter="\t")
            writer.writerow(csv_line)
        print("Question added successfully!")
    else:
        print("Question discarded.")

if __name__ == "__main__":
    main()